packages/commonjs/src/transform.js

```diff
@@ -691,5 +783,9 @@ export function transformCommonjs(
   code = magicString.toString();
   const map = sourceMap ? magicString.generateMap() : null;
 
-  return { code, map, syntheticNamedExports: isEsModule ? false : '__moduleExports' };
+  return {
+    code,
+    map,
+    syntheticNamedExports: isEsModule || isCompiledEsModule ? false : '__moduleExports'
```

Comment:
I'm avoiding creating `__moduleExports` for compiled es modules. This follows the pattern of turning the commonjs module into a real es module. I'm not sure if this can create issues in other places though.